iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java

```diff
@@ -94,4 +100,59 @@ public TimeValuePairSorter query(String deviceId, String measurement, TSDataType
     return memTableMap.get(deviceId).get(measurement);
   }
 
+  @Override
+  public void delete(String deviceId, String measurementId, long timestamp) {
+    Map<String, IWritableMemChunk> deviceMap = memTableMap.get(deviceId);
+    if (deviceMap != null) {
+      IWritableMemChunk chunk = deviceMap.get(measurementId);
+      IWritableMemChunk newChunk = filterChunk(chunk, timestamp);
+      if (newChunk != null) {
+        deviceMap.put(measurementId, newChunk);
+      }
+    }
+  }
+
+  /**
+   * If chunk contains data with timestamp less than 'timestamp', create a copy and delete all those
+   * data. Otherwise return null.
+   *
+   * @param chunk the source chunk.
+   * @param timestamp the upper-bound of deletion time.
+   * @return A reduced copy of chunk if chunk contains data with timestamp less than 'timestamp', of
+   * null.
+   */
+  private IWritableMemChunk filterChunk(IWritableMemChunk chunk, long timestamp) {
+    List<TimeValuePair> timeValuePairs = chunk.getSortedTimeValuePairList();
+    if (timeValuePairs.size() > 0 && timeValuePairs.get(0).getTimestamp() <= timestamp) {
+      TSDataType dataType = chunk.getType();
+      IWritableMemChunk newChunk = genMemSeries(dataType);
+      for (TimeValuePair pair : timeValuePairs) {
+        if (pair.getTimestamp() > timestamp) {
+          switch (dataType) {
+            case BOOLEAN:
+              newChunk.putBoolean(pair.getTimestamp(), pair.getValue().getBoolean());
+              break;
+            case DOUBLE:
+              newChunk.putDouble(pair.getTimestamp(), pair.getValue().getDouble());
+              break;
+            case INT64:
+              newChunk.putLong(pair.getTimestamp(), pair.getValue().getLong());
+              break;
+            case INT32:
+              newChunk.putInt(pair.getTimestamp(), pair.getValue().getInt());
+              break;
+            case FLOAT:
+              newChunk.putFloat(pair.getTimestamp(), pair.getValue().getFloat());
+              break;
+            case TEXT:
+              newChunk.putBinary(pair.getTimestamp(), pair.getValue().getBinary());
+              break;
+          }
```

Comment:
Add `default` and throw `UnsupportedOperationException` could remind developers of potential bugs when new `TSDataType` is added

Reply:
fixed